src/reducers/entities/channels.ts

```diff
@@ -246,18 +246,6 @@ function myMembers(state: RelationOneToOne<Channel, ChannelMembership> = {}, act
             nextState[cm.channel_id] = cm;
         }
 
-        // We remove membership from archived channels. In version 5.21 we are
-        // sending the archived channels so we have those channel synchronized
-        // and we don't need this code (MM-22201)
-        if (action.sync && action.channels) {
```

Comment:
when the websocket reconnects it will remove all channel memberships from other teams that you loaded previously.